src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -2030,296 +1406,42 @@ ObjectMonitor* ObjectSynchronizer::inflate(Thread* self, oop object,
-
-    if (next != NULL) {
-      // We lock next so that an om_flush() thread that is behind us
-      // cannot pass us when we unlock the current mid.
-      om_lock(next);
-      next_next = unmarked_next(next);
-    }
-
-    // Only try to deflate if mid is old (is not newly allocated and
-    // is not newly freed).
-    if (mid->is_old() && deflate_monitor(mid, free_head_p, free_tail_p)) {
-      // Deflation succeeded and already updated free_head_p and
-      // free_tail_p as needed. Finish the move to the local free list
-      // by unlinking mid from the global or per-thread in-use list.
-      if (cur_mid_in_use == NULL) {
-        // mid is the list head and it is locked. Switch the list head
-        // to next which is also locked (if not NULL) and also leave
-        // mid locked. Release semantics needed since not all code paths
-        // in deflate_monitor() ensure memory consistency.
-        Atomic::release_store(list_p, next);
-      } else {
-        ObjectMonitor* locked_next = mark_om_ptr(next);
-        // mid and cur_mid_in_use are locked. Switch cur_mid_in_use's
-        // next field to locked_next and also leave mid locked.
-        // Release semantics needed since not all code paths in
-        // deflate_monitor() ensure memory consistency.
-        cur_mid_in_use->release_set_next_om(locked_next);
-      }
-      // At this point mid is disconnected from the in-use list so
-      // its lock longer has any effects on in-use list.
-      deflated_count++;
-      Atomic::dec(count_p);
-      // mid is current tail in the free_head_p list so NULL terminate
-      // it (which also unlocks it). No release semantics needed since
-      // Atomic::dec() already provides it.
-      mid->set_next_om(NULL);
-
-      // All the list management is done so move on to the next one:
-      mid = next;  // mid keeps non-NULL next's locked state
-      next = next_next;
-    } else {
-      // mid is considered in-use if mid is not old or deflation did not
-      // succeed. A mid->is_new() node can be seen here when it is freshly
-      // returned by om_alloc() (and skips the deflation code path).
-      // A mid->is_old() node can be seen here when deflation failed.
-      // A mid->is_free() node can be seen here when a fresh node from
-      // om_alloc() is released by om_release() due to losing the race
-      // in inflate().
-
-      // All the list management is done so move on to the next one:
-      if (cur_mid_in_use != NULL) {
-        om_unlock(cur_mid_in_use);
-      }
-      // The next cur_mid_in_use keeps mid's lock state so
-      // that it is stable for a possible next field change. It
-      // cannot be modified by om_release() while it is locked.
-      cur_mid_in_use = mid;
-      mid = next;  // mid keeps non-NULL next's locked state
-      next = next_next;
+      // Honor block request.
+      ThreadBlockInVM tbivm(self->as_Java_thread());
```

Comment:
Same comment as previous use of TBIVM. (It's hard to see in the PR UI how they two blocks relate.)

Reply:
This kind of use of ThreadBlockInVM predates this changeset so while
the location is new, then code style is old, very old... I'll hold off changing
this for now.